src/co2.js

```diff
@@ -252,14 +248,22 @@ class CO2 {
     return all;
   }
 
+  /**
+   *
+   * @param {PageXRay} pageXray
+   * @param {string[]=} greenDomains
+   */
   dirtiestResources(pageXray, greenDomains) {
+    /** @type {CO2PerContentAsset[]} */
     const allAssets = [];
     for (let asset of pageXray.assets) {
-      const domain = new URL(asset.url).domain;
+      const domain = new URL(asset.url).host;
```

Comment:
question: is this change correct? typescript doesn't recognize that there is a `domain` property of a `URL` instance, but it's possible it's inferring the wrong `URL` type